js/server/modules/@arangodb/foxx/manager.js

```diff
@@ -848,22 +830,35 @@ function install (serviceInfo, mount, options = {}) {
       `
     });
   }
-  const tempPaths = _prepareService(serviceInfo, options);
-  _install(mount, tempPaths, options);
+  const tempPaths = _prepareService(serviceInfo, options.legacy);
+  const tempService = FoxxService.create({
+    mount,
+    basePath: tempPaths.tempServicePath,
+    noisy: true,
+    options: {
+      development: options.development,
+      configuration: options.configuration,
+      dependencies: options.dependencies
+    }
+  });
+  _install(tempService, tempPaths.tempBundlePath, {
+    setup: options.setup !== false,
+    force: options.force === true
+  });
   selfHeal();
   propagateSelfHeal();
+  reloadRouting();
 
-  return FoxxService.create({
-    mount,
-    options
-  });
+  return getServiceInstance(mount);
```

Comment:
Unlike the `FoxxService.create` this actually ensures we're returning a service instance that's actually live and handling requests. We're also potentially skipping yet another redundant validation.